src/Native/Runtime/amd64/StubDispatch.S

```diff
@@ -0,0 +1,171 @@
+        save_xmm128_postrsp xmm5, (30h + 50h + 16*5)
+        save_xmm128_postrsp xmm6, (30h + 50h + 16*6)
+        save_xmm128_postrsp xmm7, (30h + 50h + 16*7)
+        END_PROLOGUE
+
+        // Build PInvokeTransitionFrame. This is only required if we end up resolving the interface method via
+        // a callout to a managed ICastable method. In that instance we need to be able to cope with garbage
+        // collections which in turn need to be able to walk the stack from the ICastable method, skip the
+        // unmanaged runtime portions and resume walking at our caller. This frame provides both the means to
+        // unwind to that caller and a place to spill callee saved registers in case they contain GC
+        // references from the caller.
+
+        // Save caller's rip.
+        mov     rax, [rsp + RIDS_ReservedStack]
+        mov     [rsp + 30h + 8*0], rax
+
+        // Save caller's rbp.
+        mov     [rsp + 30h + 8*1], rbp
+
+        // Zero out the Thread*, it's not used by the stackwalker.
+        xor     rax, rax
+        mov     [rsp + 30h + 8*2], rax
+
+        // Set the flags.
+        mov     dword ptr [rsp + 30h + 8*3], PTFF_SAVE_ALL_PRESERVED + PTFF_SAVE_RSP
+
+        // Save callee saved registers.
+        mov     [rsp + 30h + 8*4], rbx
+        mov     [rsp + 30h + 8*5], r12
+        mov     [rsp + 30h + 8*6], r13
+        mov     [rsp + 30h + 8*7], r14
+        mov     [rsp + 30h + 8*8], r15
+
+        // Calculate and store the caller's rsp.
+        lea     rax, [rsp + RIDS_ReservedStack + 8]
+        mov     [rsp + 30h + 8*9], rax
+
+        // First argument is the instance we're dispatching on which is already in rdi.
+
+        // Second argument is the dispatch data cell. We still have this in r10
+        mov     rsi, r10
+
+        // The third argument is the address of the transition frame we build above.
+        lea     rdx, [rsp + 30h]
+
+        call    C_FUNC(RhpResolveInterfaceMethodCacheMiss)
+
+        // Recover callee-saved values from the transition frame in case a GC updated them.
+        mov     rbx, [rsp + 30h + 8*4]
+        mov     r12, [rsp + 30h + 8*5]
+        mov     r13, [rsp + 30h + 8*6]
+        mov     r14, [rsp + 30h + 8*7]
+        mov     r15, [rsp + 30h + 8*8]
+
+        // Restore the argument registers.
+        movdqa  xmm0, [rsp + 30h + 50h + 16*0]
+        movdqa  xmm1, [rsp + 30h + 50h + 16*1]
+        movdqa  xmm2, [rsp + 30h + 50h + 16*2]
+        movdqa  xmm3, [rsp + 30h + 50h + 16*3]
+        movdqa  xmm4, [rsp + 30h + 50h + 16*4]
+        movdqa  xmm5, [rsp + 30h + 50h + 16*5]
+        movdqa  xmm6, [rsp + 30h + 50h + 16*6]
+        movdqa  xmm7, [rsp + 30h + 50h + 16*7]
+        mov     r9,   [rsp + 8*5]
+        mov     r8,   [rsp + 8*4]
+        mov     rcx,  [rsp + 8*3]
+        mov     rdx,  [rsp + 8*2]
+        mov     rsi,  [rsp + 8*1]
+        mov     rdi,  [rsp + 8*0]
+
+        .att_syntax
+        add     $RIDS_ReservedStack, %rsp
```

Comment:
A nit - you can use #define instead of .equ and then you don't need to switch the syntax.